### DISCUSSION TITLE
Help with conditional rules

### DISCUSSION THREAD

**Discussion by DarthWeber** [arendst/Tasmota#22557] [Support request]
Hi, i am stucked in an idea and needs some help with the following Problem. My rule is:

> ON PCF8574-1_INP#D0=0 DO backlog Pulsetime1 160; Pulsetime3 0; Power3 ON; Delay 20; Power1 toggle ENDON ON PCF8574-1_INP#D1=0 DO Pulsetime2 160; Pulsetime3 0; Power3 ON; Delay 20; Power2 toggle ENDON

i have a valve on relay 1 and the power supply on relay 3 which i switch off during winter or night. To avoid starting issues i switch on relay 3 first and after 2 seconds the valve. That works fine.

But I want to omit the delay if relay3 is on already. Since i also use rule2 and rule3 it would be great to get it in the same rule, maybe it is possible to calculate the delay time with the value of relay3 like "delay 20*power3" ? I read that the if statements does not work with the backlog.

Many thanks,
Tom

**Comment by Noschvie** [comment]
if you use an ESP32 you can use Berry instead of rules.
